### PR TITLE
feat: prep work for Cognito SMS Sandbox #2

### PR DESCRIPTION
**Pull request by yuth**
#### Description of changes

This change enables Amplify CLI add support for querying the SNS SMS Sandbox status. The SDK changes are still not released but when the SDK change is available, CLI should be able to query the sandbox status.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-provider-awscloudformation/src/__tests__/display-helpful-urls.test.ts [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+      });
+
+      await showSMSSandboxWarning(context);
+
+      expect(mockedGetMessage).toHaveBeenCalledWith('COGNITO_SMS_SANDBOX_MISSING_PERMISSION');
+      expect(context.print.warning).toHaveBeenCalledWith(message);
+    });
+  });
+
+  describe('it should not show any warning  message when the SNS API is not deployed', () => {
```
Comment (cjihrig) [inline packages/amplify-provider-awscloudformation/src/__tests__/display-helpful-urls.test.ts:82]:
```suggestion
  describe('it should not show any warning message when the SNS API is not deployed', () => {
```

File: packages/amplify-provider-awscloudformation/src/aws-utils/aws-sns.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  }
+
+  private constructor(context: $TSContext, cred: $TSAny, options = {}) {
+    this.sns = new aws.SNS({ ...cred, ...options });
+  }
+
+  public async isInSandboxMode(): Promise<boolean> {
+    // AWS SDK still does not have getSMSSandboxAccountStatus. Casting sns to any to avoid compile error
+    // Todo: remove any casting once aws-sdk is updated
+    this.sns.createTopic().promise();
```
[the remark this answers is not in the document]
Comment (yuth) [reply packages/amplify-provider-awscloudformation/src/aws-utils/aws-sns.ts:30]: This was some leftover code. I will remove it

File: packages/amplify-provider-awscloudformation/src/aws-utils/aws-sns.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  }
+
+  private constructor(context: $TSContext, cred: $TSAny, options = {}) {
+    this.sns = new aws.SNS({ ...cred, ...options });
+  }
+
+  public async isInSandboxMode(): Promise<boolean> {
+    // AWS SDK still does not have getSMSSandboxAccountStatus. Casting sns to any to avoid compile error
+    // Todo: remove any casting once aws-sdk is updated
+    const snsClient = (this.sns as unknown) as any;
```
Comment (cjihrig) [inline packages/amplify-provider-awscloudformation/src/aws-utils/aws-sns.ts:30]: Do you need to cast to unknown before casting to any?

**Review by cjihrig — APPROVED** [review]

File: packages/amplify-provider-awscloudformation/src/aws-utils/aws-sns.ts (hunk 3 above)
Comment (yuth) [reply packages/amplify-provider-awscloudformation/src/aws-utils/aws-sns.ts]: no, its needed when the type of the variable is not defined initally I think

**Review by SwaySway — APPROVED** [review]
aside from nit LGTM

**Pull request merged**